src/org/opendatakit/briefcase/ui/reused/source/ShowSourceForm.java

```diff
@@ -32,26 +32,44 @@
   private JLabel sourceLabel;
   private JButton resetButton;
   private JLabel actionLabel;
+  private JButton reloadButton;
   private List<Runnable> onResetCallbacks = new ArrayList<>();
+  private List<Runnable> onReloadCallbacks = new ArrayList<>();
+  private Operation type;
 
-  private ShowSourceForm(String action) {
+  private ShowSourceForm(String action, Operation type) {
     $$$setupUI$$$();
     this.action = action;
+    this.type = type;
 
     resetButton.addActionListener(__ -> onResetCallbacks.forEach(Runnable::run));
+    if (type.equals(Operation.PULL)) {
+      reloadButton.addActionListener(__ -> onReloadCallbacks.forEach(Runnable::run));
+    } else {
+      reloadButton.setVisible(false);
+    }
   }
 
-  static ShowSourceForm empty(String action) {
-    return new ShowSourceForm(action);
+  static ShowSourceForm pull(String action) {
+    return new ShowSourceForm(action, Operation.PULL);
+  }
+
+  static ShowSourceForm push(String action) {
+    return new ShowSourceForm(action, Operation.PUSH);
   }
 
   void onReset(Runnable callback) {
     onResetCallbacks.add(callback);
   }
 
+  void onReload(Runnable callback) {
+    onReloadCallbacks.add(callback);
+  }
+
   void showSource(Source source) {
     actionLabel.setText(action + ": " + source.toString());
     sourceLabel.setText(source.getDescription());
+    reloadButton.setVisible(source instanceof Source.Aggregate);
```

Comment:
By checking the subtype of `Source` we're breaking the encapsulation of this class hierarchy. As an alternative, we could add a `boolean canBeReloaded()` method to the `Source` interface.

Also, is this missing to check the `Operation`? It seems like it would show the button on the push tab too when the target is an Aggregate server.

Reply:
Am I right in understanding that by adding a `boolean canBeReloaded()` method to the `Source` interface we'd maintain encapsulation by leaving the decision to support `Reload` up to the specific `Source`, in this case, `Source.Aggregate`?

You're right about how it's missing a check on the value of `Operation`. I've since switched `Operation` to be a boolean and have checked it's value here.

Reply:
Yup, that was my rationale :)